datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1145,6 +1145,22 @@ impl Projection {
         })
     }
 
+    /// Create a new Projection using the specified output schema
+    pub fn new_from_schema(input: Arc<LogicalPlan>, schema: DFSchemaRef) -> Self {
+        let expr: Vec<Expr> = schema
+            .fields()
+            .iter()
+            .map(|field| field.qualified_column())
+            .map(Expr::Column)
+            .collect();
+        Self {
+            expr,
+            input,
+            schema,
+            alias: None,
+        }
+    }
```

Comment:
I'm not sure, if the schema here has different names with the input plan's schema, do we need to create an Alias Expr or a normal Column Expr?

Reply:
Yes, your concerns are justified

Reply:
Your concerns are justified, fix it. 

I noticed it before, but I forgot when I was writing the code

BTW, I think we should add alias in Union, how do you think it? We should remove union alias, and replace with projection. @mingmwang

Reply:
```scala
spark Union

case class Union(
    children: Seq[LogicalPlan],
    byName: Boolean = false,
    allowMissingCol: Boolean = false) extends LogicalPlan {
}
```

Reply:
> Your concerns are justified, fix it.
> 
> I noticed it before, but I forgot when I was writing the code
> 
> BTW, I think we should add alias in Union, how do you think it? We should remove union alias, and replace with projection. @mingmwang

I think the Alias in Datafusion Union/Projection is more like the SubqueryAlias in SparkSQL, it is not the Column level Alias Expr. If the original Union has the alias, you can keep the alias in the Union and no need to move alias from Union to Projection.  But if the Union is eliminated(only 1 input), we should keep the Alias in Projection.  
I think besides checking the Union Alias, you also need to check whether we should add additional Alias in new projection exprs.

````
let new_plan = {
new children len ==  1  and there is alias in original Union: 
    Projection(child, alias)
new children  len ==  1 input and no alias in original Union
    child
new children len > 1
   let new_union_schema = from_new_children(new_children)
    Union(new_children, new_union_schema, union.alias);
}

if new_plan.schema != union.schema {
  // Add new projection for column level Alias maybe
}
````

Reply:
when `new children  len ==  1`, we must add projection, because union schema != child schema.

Reply:
Make sense to me.
But I think add projection directly also is valid.

Reply:
But this might introduce unnecessary Projection. I would suggest to add Projection only when it is really unnecessary.

Reply:
Make sense👍